system-functionalities.md

```diff
@@ -132,7 +132,10 @@ Protocol:
 1. The client:
     1. Runs the [generate](#generate-a-secret) protocol on input `(32, rng, user_id||key_id||"client-generated")` to get a secret `arbitrary_key`.
     1. Computes `ciphertext = Enc(storage_key, arbitrary_key, user_id||key_id||"client-generated")` and sends `ciphertext` to the key server over the secure channel.
-    1. [Stores](#client-side-storage) `arbitrary_key`and associated data `user_id||key_id"||"client-generated"` locally.
+    1. [Prepares to store](#client-side-storage) `arbitrary_key`and associated data `user_id||key_id"||"client-generated"` locally by creating a storage object to be returned to the calling application at the end of the protocol. This storage object, `StorageObject`, should contain `arbirary_key` and the associated data `user_id||key_id"||"client-generated"`.
```

Comment:
Is the associated data here different from what is included in the [`Secret`](https://github.com/boltlabs-inc/key-mgmt/blob/develop/dams/src/crypto.rs#L515) type already?

Reply:
@marsella Can you confirm? I believe in the implementation, the associated data here is folded into the `Secret` type.

Reply:
Yes, the secret type holds associated data, and we use exactly this ad in [the public function to create and encrypt a secret](https://github.com/boltlabs-inc/key-mgmt/blob/develop/dams/src/crypto.rs#L373-L390).